fabric/src/main/java/net/blay09/mods/balm/mixin/MinecraftMixin.java

```diff
@@ -40,4 +44,14 @@ public void startUseItem(CallbackInfo callbackInfo, InteractionHand[] hands, int
         }
     }
 
+    @Inject(method = "clearLevel(Lnet/minecraft/client/gui/screens/Screen;)V", at = @At("HEAD"))
+    public void clearLevel(Screen p_91321_, CallbackInfo ci) {
+        if (this.level != null) {
+            Balm.getEvents().fireEvent(new LevelEvent.Unload(this.level));
+        }
+    }
+
+
```

Comment:
Forge also fires Unload for the previous level when `setLevel(Lnet/minecraft/client/multiplayer/ClientLevel;)V` is called, should probably do the same here for parity